services/dynamic-sidecar/tests/unit/test_api_containers_tasks.py

```diff
@@ -0,0 +1,604 @@
+    ]
+
+    original_tasks: dict[str, Callable] = {}
+
+    for task_name in tasks_names:
+        original_tasks[task_name] = getattr(containers_tasks, task_name)
+        setattr(containers_tasks, task_name, _just_log_task)
+
+    yield
+
+    for task_name, original_task in original_tasks.items():
+        setattr(containers_tasks, task_name, original_task)
+
+
+@asynccontextmanager
+async def auto_remove_task(client: Client, task_id: TaskId) -> AsyncIterator[None]:
+    """clenup pending tasks"""
+    try:
+        yield
+    finally:
+        await client.cancel_and_delete_task(task_id, timeout=10)
+
+
+async def _get_container_timestamps(
+    container_names: list[str],
+) -> dict[str, ContainerTimes]:
+    container_timestamps: dict[str, ContainerTimes] = {}
+    async with aiodocker.Docker() as client:
+        for container_name in container_names:
+            container: DockerContainer = await client.containers.get(container_name)
+            container_inspect: dict[str, Any] = await container.show()
+            container_timestamps[container_name] = ContainerTimes(
+                created=container_inspect["Created"],
+                started_at=container_inspect["State"]["StartedAt"],
+                finished_at=container_inspect["State"]["FinishedAt"],
+            )
+
+    return container_timestamps
+
+
+# FIXTURES
+
+
+@pytest.fixture
+def dynamic_sidecar_network_name() -> str:
+    return _get_dynamic_sidecar_network_name()
+
+
+@pytest.fixture(
+    params=[
+        {
+            "version": "3",
+            "services": {
+                "first-box": {
+                    "image": "busybox",
+                    "networks": [
+                        _get_dynamic_sidecar_network_name(),
+                    ],
+                },
+                "second-box": {"image": "busybox"},
+            },
+            "networks": {_get_dynamic_sidecar_network_name(): {}},
+        },
+        {
+            "version": "3",
+            "services": {
+                "solo-box": {"image": "busybox"},
+            },
+        },
+    ]
+)
+def compose_spec(request: FixtureRequest) -> str:
+    spec_dict: dict[str, Any] = request.param  # type: ignore
+    return json.dumps(spec_dict)
+
+
+@pytest.fixture
+def backend_url() -> AnyHttpUrl:
+    return parse_obj_as(AnyHttpUrl, "http://backgroud.testserver.io")
+
+
+@pytest.fixture
+async def app(app: FastAPI) -> AsyncIterable[FastAPI]:
+    # add the client setup to the same application
+    # this is only required for testing, in reality
+    # this will be in a different process
+    client_setup(app)
+    async with LifespanManager(app):
+        __print_routes(app)
+        yield app
+
+
+@pytest.fixture
+async def httpx_async_client(
+    app: FastAPI,
+    backend_url: AnyHttpUrl,
+    ensure_external_volumes: tuple[DockerVolume],
+    cleanup_containers: None,
+) -> AsyncIterable[AsyncClient]:
+    async with AsyncClient(
+        app=app,
```

Comment:
don't we have this aync client already in a conftest or so?

Reply:
please have a look at services/api-server/tests/unit/conftest.py

Reply:
forgotten?

Reply:
Sorry did not finish checking this. It's present in other services, not in `pytest_simcore`.